src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -182,10 +193,10 @@ protected static QueryParseContext createContext() {
         return new QueryParseContext(index, queryParserService);
     }
 
-    private static void assertQueryHeader(XContentParser parser, String expectedParserName) throws IOException {
+    private void assertQueryHeader(XContentParser parser) throws IOException {
         assertThat(parser.nextToken(), is(XContentParser.Token.START_OBJECT));
         assertThat(parser.nextToken(), is(XContentParser.Token.FIELD_NAME));
-        assertThat(parser.currentName(), is(expectedParserName));
+        assertThat(parser.currentName(), is(testQuery.parserName()));
```

Comment:
super minor nitpick: the parserName() is there only temporarily I think? Maybe we shouldn't use it then so it is going to be easier to remove it?